parsons/google/google_sheets.py

```diff
@@ -395,6 +399,148 @@ def format_cells(self, spreadsheet_id, range, cell_format, worksheet=0):
+            max: int
+                How many attempts to make before giving up - defaults to 4
+            wait_time: int
+                Number of seconds to wait between attempts - defaults to 15
+            kwargs: dict
+                Any arguments required by `method` - note that positional args will have to be named
+        `Returns:`
+            Whatever `method` is supposed to return
+
+        """
+
+        # Recursively account for nested methods as needed
+        nested_methods = method.split(".")
+
+        if len(nested_methods) == 1:
+            final_method = self
+        else:
+            final_method = self[nested_methods[0]]
+            nested_methods.pop(0)
+
+        try:
+
+            # If final_method isn't callable, then the API call is made in the loop, not below
+            for m in nested_methods:
+                final_method = getattr(final_method, m)
+
+            # Using getattr allows the method/attribute to be user-provided
+            if callable(final_method):
+                output = final_method(**kwargs)
+            else:
+                output = final_method
+
+        except (APIError, HTTPError, ReadTimeout, ConnectionError) as e:
+            # Lets get the ordinals right, because why not
+            if i % 10 == 1:
+                ordinal = "st"
+            elif i % 10 == 2:
+                ordinal = "nd"
+            else:
+                ordinal = "th"
+
+            logger.info(f"trying to {method} for the {i}{ordinal} time")
+            if i < max:
+                time.sleep(wait_time)
+                i += 1
+                output = self.attempt_gsheet_method(method, i, max, wait_time, **kwargs)
+
+            else:
+                raise e
+
+        return output
+
+    def combine_multiple_sheet_data(self, sheet_ids, worksheet_id=None):
+        """
+        Combines data from multiple Google Sheets into a Parsons Table.
+        `Args:`
+            sheet_ids: str, list
+                The IDs of the Google Spreadsheets with that data to be combined. Can be a
+                comma-separated string of IDs or a list.
+
+            worksheet_id: str
+                If None, the first worksheet (ID = 0) is assumed.
+
+        `Returns:` obj
+            Parsons Table containing the concatenated data from all the sheets.
+        """
+        id_col = "sheet_id"
+        sheet_id_list = []
+
+        # Parse different possible sheet_ids types
+        if isinstance(sheet_ids, list):
+            # Already a list!
+            sheet_id_list = sheet_ids
+
+        elif "," in sheet_ids:
+            # Comma-separated string
+            sheet_id_list = [x.strip() for x in sheet_ids.split(",")]
+
+        else:
+            raise ValueError(f"{sheet_ids} is not a valid string or list GSheet IDs")
+
+        # Non-DB table options yield a list, convert to Parsons table with default worksheet col
+        if sheet_id_list:
+            sheet_id_tbl = Table(
```

Comment:
What is the benefit of making this a table instead of a list of dictionaries?

Reply:
Could it just be `sheet_id_tbl = [{"sheet_id": x, "worksheet_id": 0} for x in sheet_id_list]`? I think that would even work as-is. (Although we'd probably want to change the name...)